### DISCUSSION TITLE
How can I restore old `no_gaps_when_only` behavior for special workspaces?

### DISCUSSION THREAD

**Discussion by tealsnow** [hyprwm/Hyprland#8530] [Q&A] [answered]
It was that special workspaces were unaffected by `no_gaps_when_only`. I liked this as it was an easy way to visually tell if I was using a special workspace or not. How can I recreate this behavior using the new rules?

**Comment by vaxerski** [comment]
RTFM https://wiki.hyprland.org/Configuring/Workspace-Rules/#smart-gaps

**Comment by tealsnow** [comment]
Maybe I don't understand the F***ing Manual.
That, or should I open a bug report? Because everything I've tried does not work.
Maybe this is why I'm asking for help in the first place.

**Reply by vaxerski** [reply, marked as answer]
~~have you put the rules in your config?~~

Oh wait, I misread your comment. You can add `s[0]` to the workspace selectors to target non-special workspaces. rtfm moment anyways

**Reply by tealsnow** [reply]
Yep that worked. Appreciate the help.
~~For reference, the exact config that worked was:~~
```
workspace = s[0], gapsout:0, gapsin:0
windowrulev2 = bordersize 0, floating:0, onworkspace:s[0]
windowrulev2 = rounding 0, floating:0, onworkspace:s[0]
```

Actually, just the above makes gaps *never* appear for normal workspaces.
The below achieves no gaps when there is a single window on a normal workspace, does have gaps when there are more than one window on a normal workspace and gaps always in special workspaces:
```
workspace = w[t1], gapsout:0, gapsin:0
windowrulev2 = bordersize 0, floating:0, onworkspace:w[t1]
windowrulev2 = rounding 0, floating:0, onworkspace:w[t1]
workspace = w[tg1], gapsout:0, gapsin:0
windowrulev2 = bordersize 0, floating:0, onworkspace:w[tg1]
windowrulev2 = rounding 0, floating:0, onworkspace:w[tg1]
workspace = f[1], gapsout:0, gapsin:0
windowrulev2 = bordersize 0, floating:0, onworkspace:f[1]
windowrulev2 = rounding 0, floating:0, onworkspace:f[1]
workspace = s[1], gapsout:2, gapsin:2
windowrulev2 = bordersize 1, floating:0, onworkspace:s[1]
windowrulev2 = rounding 0, floating:0, onworkspace:s[1]
```
